tools/assetpipelineeditor/exportdialog.cpp

```diff
@@ -0,0 +1,104 @@
+#include "exportdialog.h"
+#include "ui_exportdialog.h"
+
+#include <Kuesa/private/gltf2exporter_p.h>
+
+#include <QFile>
+#include <QFileInfo>
+#include <QDir>
+#include <QJsonDocument>
+#include <QFileDialog>
+#include <QStyle>
+#include <QPushButton>
+#include <QMessageBox>
+
+ExportDialog::ExportDialog(Kuesa::GLTF2Exporter &exporter, const QString &path, QWidget *parent)
+    : QDialog(parent)
+    , ui(new Ui::ExportDialog)
+    , m_exporter(exporter)
+    , m_originalFile(path)
+    , m_targetFile(m_originalFile)
+{
+    ui->setupUi(this);
+
+    ui->sourceLabel->setText(tr("Saving: ") + m_originalFile);
+
+    auto save_btn = ui->buttonBox->button(QDialogButtonBox::Save);
+    connect(save_btn, &QPushButton::clicked,
+            this, &ExportDialog::onSave);
+}
+
+ExportDialog::~ExportDialog()
+{
+    delete ui;
+}
+
+void ExportDialog::onSave()
+{
+    ui->errorLog->setPlainText("");
+
+    m_targetFile = QFileDialog::getSaveFileName(this, tr("Select a folder to save to"), m_targetFile, tr("glTF 2.0 file (*.gltf)"));
```

Comment:
We're not handling the case where the user clicked cancel and we got an empty string back. We should likely bail out early then. Would be even better if that was selected from within the original dialog instead of a file dialog on top of a dialog when clicking save.

Reply:
is the best way to go towards this is to make the ExportDialog inherit from QFileDialog and add the compression controls at the bottom ? (and is it ok if it's not the native file dialogs anymore in that case ?)

Reply:
also it used to be like this before, should I maybe revert to that ? https://user-images.githubusercontent.com/2772730/51386615-57104480-1b23-11e9-8095-e198d147c6b7.png

Reply:
Yes, I think I preferred that older approach indeed. Sorry for the rework incurred.

Reply:
old approach had the issue that it didn't warm the user about overwriting the original file

Reply:
Clearly it should warn of course, this is unrelated to how the GUI though.

Reply:
![mesh](https://user-images.githubusercontent.com/2772730/52701296-300a2e80-2f7a-11e9-9d15-3df0c7eec5ea.png) here's how it looks now - the icon on the lineedit opens the save dialog. Is it ok ? (it seemed clearer to me than the previous solution)